### ISSUE TITLE
Collective admins can create pending contribution.

### ISSUE THREAD

**Issue by natehn** [opencollective/opencollective#5769]
## User story 
As an OCF host admin, I get tons of emails from Collectives about incoming bank transfers and checks. I am always digging around my email and the help desk (and even Slack) to try and match things up. Only a small portion of bank transfers are logged as pending by the donor in the platform, and even if we bring bank transfers into the platform, as I have [previously suggested](https://github.com/opencollective/opencollective/issues/4927), many donors, especially large funders, will not engage with the platform.

## Best solution for this problem
Allowing Collectives, who often know when funds are coming, to create pending contributions. This is something we had at Fractured Atlas that made the process much smoother. Host admins will need to be able to manipulate pending contributions, as mentioned in https://github.com/opencollective/opencollective/issues/5087.

## MVP
I'll leave this part up to you all. :)

<details>
  <summary>Triage Template (team only)</summary>

  This template is for members of the team to triage for prioritisation. For more guidance see https://www.loom.com/share/369ab467fbc64dec848085d38ff57ca0:

  P1 high frequency, high impact
  P2 low frequency, high impact
  P3 high frequency, low impact
  P4 low frequency, low impact

  Examples of high impact - a problem affects users during an essential process (expenses and payments > onboarding and registration > contributing) with no workaround.

  High frequency - >10% of users affected (measured as a proportion of total potential users for this case).
</details>

**Comment by natehn** [comment]
One thing that could make this even _more_ amazing if it helped track basic check information. This would require an additional field, I think: the check number, likely added by the host admin once it is received and/or deposited (this could be helpful for receiving check donations in general - a field where the donor includes a check number when sending the check and it creates a pending donation)

The other thing that would be essential is improved searching for pending donations. By Collective, for example, and being able to search for a specific donation amount.

**Comment by iamronen** [comment]
@poohlaga can you please chime in on this: this strikes me as familiar to something we discussed yesterday; the existing capability to create payment orders in the platform?

**Comment by natehn** [comment]
There should also be a field for special instructions, and the ability to create incognito donations (which relates to https://github.com/opencollective/opencollective/issues/5453)

**Comment by natehn** [comment]
In practice, this would be very helpful for one of the essential functions of fiscal hosts, receiving funds. Right now, we have tons and tons of "pending" tickets and random email threads, and we sift through all of them every single time. This leads to mistakes, such as accidentally making an anonymous donation public, or sending funds to the wrong Collective. 

Being able to take emailed check or bank transfer info and create the pending donation would alleviate many of these issues, and remove the need for huge spreadsheets of checks and lists of missing contributions. It would be an absolute game changer!

**Comment by natehn** [comment]
This also builds on an existing feature - so I wonder how time-intensive it will be?

**Comment by BenJam** [comment]
don't we already have this through pending orders https://docs.opencollective.com/help/independent-collectives/manage-independent-collective/money-coming-in-contributions#via-bank-transfer ?

I think @alinamanko used to use this more, we looked into extending the age out for pending transfers from 30 days (current) to 60 or 90 at Open Source Collective. 

@znarf can i request a fact check?

**Comment by natehn** [comment]
@BenJam this would be an expansion of that feature

**Comment by natehn** [comment]
Sometimes:

1. The bank transfer does what they are supposed to, and they submit a bank transfer through the Collective's profile. But they are unlikely to do this for a check.

That said, a lot of the time:

2. Donors don't tell us, instead only communicating with Collectives. Our documentation asks, in that situation, for the Collective to tell us by email.
3. Certain big funders will never take the time to go through our system - they already have complex systems set up on their end that they have us or our Collectives go through. So they send us a notification in some weird form, and every single one is different.

So, there are three potential levels to this, in reverse order of importance:

1. Maybe: adjusting the bank transfer contribution flow to also work for checks or other income sources, and to maybe make clear that the bank transfer isn't actually happening on the platform earlier on the flow. This may not be necessary, but something to consider?
2. Probably: creating a flow for Collectives to enter info that they have for incoming donations.
3. Definitely (the MVP): adding a flow for host admins to enter all the info they would add for "added funds" but without actually adding the funds, which I assume - but it doesn't have to be this way - would then be in the pending contributions page.

**Comment by aomuse** [comment]
Would it create too much complexity to think if this could also be where we address the invoice challenges? Invoices represent another type of pending contribution which the fiscal host generates in their non-OC bookkeeping system and track until funds arrive. The ability to add a pending invoice payment which sent out a custom payment option email would remove one more off-platform step.

**Comment by alanna** [comment]
I had previously suggested adding checks as a payment method in #4024 but that was not picked up because it was pointed out, rightly, that if we publicly show checks as an options we will get more checks, which we don't want. But if the host admin could define the payment method as check when confirming the funds, that would help a lot with accounting later.

**Comment by natehn** [comment]
Wondering whether this could make this easier to build? Idk - not an engineer, but seems like we might not have to develop everything from the ground up. https://stripe.com/invoicing

**Comment by Betree** [comment]
@natehn we already have a working invoice generation system, being able to emit invoices before the payment would not be a complex addition. Also, using Stripe's invoice service to handle that would lock us with them as the only payment provider available for this feature.

**Comment by natehn** [comment]
@Betree thanks! Makes complete sense - I figured I'd throw it in there, as someone shared it with me recently

**Comment by natehn** [comment]
A note: when confirming a pending contribution, the host admin should be able to add an internal memo, as they do when adding funds: https://www.loom.com/share/d9c4a31ed365493583e80deeca4d392a

**Comment by alanna** [comment]
See #4497 regarding attaching payable invoices to pending contributions - it would be a great addition and would incentivise contributors who otherwise would be manual to use the platform.

**Comment by shannondwray** [comment]
Has a service like Poli been explored? this sends users to a payment portal which they can complete the bank transfer? https://www.polipayments.com/index.html Might only be used in Australia and New Zealand, unsure

**Comment by natehn** [comment]
@shannondwray I think this would be something to discuss in the bank transfer issue rather than this pending contributions issue - they are related, but split

**Comment by natehn** [comment]
Note: internal discussion of this issue is happening here: https://www.loomio.com/d/73ajeYAr/pending-contributions

**Comment by BenJam** [comment]
@iamronen @natehn before progressing this to development I'd like to explore a solution that utilises a lot of the work that we already have in pending contributions and invited expenses. 

There's a lot of work in the proposal and I think we can reappropriate the process we have for an invited expense, linked to a contribution, along with some improvements to the host's financial contributions dashboard.

**Comment by BenJam** [comment]
Just dropping in to leave my own take on yesterday's 'kick off' conversation.

From a practical perspective, we talked about the overlap with 'invited expenses' and the existing 'pending' flow for bank transfer contributions. We decided we would be better off basing our work on the contribution rather than the expense flow at this point despite the fact that the expense flow would come with some benefits like comments, invoices etc. 

We also spoke about the fact that this is a process between collective and host. Contributors rarely (if ever) get involved on this process as they're predominantly acting on their own terms. As such there is not really an 'invitation' to a contributor to acknowledge of approve. Instead hosts create organisations representing the contributor on their behalf. 

### Immediate priorities

To address the lack of information available to collectives and hosts about contributions that are expected, added by the host (in the process of say, signing a funding agreement) or by the Collective's administrators (say, in the process of working with an existing funder on a large contribution). i.e.

- create a process for adding a pending contribution on behalf of a donor as a host or collective (including necessary notifications) including any payment references needed
- show pending contributions managed through this process on a collective's profile page
- improve the filters available to hosts to find pending contributions 

### For later

We acknowledged that, ideally, we would want to bring some of the process that's happening outside of Open Collective into the platform inc.

- generating invoices
- adding notes and documents (for instance and funding agreement)
- conversation between host and collective admins

we thought that, once the core information is available, we could do this by adding a contribution page similar to an expense page for these kinds of contributions (or indeed, for any contribution on the platform).

**Comment by iamronen** [comment]
Yesterday in a scoping conversation attended by @kewitz @Memo-Es @znarf and myself we outlined a 3 phase plan for approaching this project.

attn: @natehn 

### Phase1: Pending Contributions for Fiscal Hosts

1. Expanding the existing pending contributions capability (originally created for bank transfers) to enable Fiscal Host admins to:
    - Create pending contributions with the information needed to track and reconcile them when funds arrive and are reconciled.
    - Track pending contributions and search them for matching against incoming funds.
    - Add funds through the pending contribution.
3. Expose to **collective admins pending contributions** that are associated with their collectives.
4. Notifications:
    - to collective - pending order created
    - to collective - automated reassurance "we are tracking this" 
    - to collective - funds arrived
    - to funder contact - funds received

This will enable fiscal hosts to channel all funding related information onto the platform where they can be tracked and reconciled. This addresses the primary pain points. This is the scope of work for Q1C1 2023.

### Phase2: Collectives Create Pending Contributions

Once established for fiscal hosts, the capability will be extended to collectives so that they can independently create pending contributions as they do their own fundraising.

This will both empower collectives and relieve work from fiscal hosts. Collectives will be able to independently and proactively notify Fiscal Hosts to expect incoming funds. Fiscal host admins will not need to do know about or engage with pending contributions until explicit action is required from them (eg: contract agreement or reconciling incoming funds).

### Phase3: Apply the Conversational Work Pattern

The "conversation" UX pattern that is currently available for expenses (which enables a comment based conversation thread with attached documents) will be applied to pending contributions to fully encapsulate the fundraising coordination process between collectives and fiscal hosts.

Hopefully, this will converge with the maturity of the "workplace" transformation which will include in-app notifications which, ideally, will enable all communications regarding pending contributions will take place on the platform with no need to use freshdesk, email, slack or any other tool. Fiscal hosts (who choose to utilize this capability) will be able to direct collective who fund-raise for the first time to this platform capability.

**Comment by iamronen** [comment]
@kewitz my understanding is that our current working assumption that we are NOT touching invoices. We agreed that this is a desirable capability (removes the manual copy-pasting between screens) but needs to be addressed systemically and has legal ramifications.

One possibility that came up was generating an interface with (for example) Quickbooks to automatically generate invoices (in Quickbooks) based on the information we (will) already have in the (new) pending contribution. This is currently not in scope because the working assumption is that it is a complicated and expensive request. Looking into this in the future will require an explicit decision and prioritization to do so.

**Comment by iamronen** [comment]
since we are not including in this phase the "conversational work pattern" we need to consider adding:

1. Purchase Order#
2. Notes field (that in the future can be transitioned into the conversational pattern).

**Comment by BenJam** [comment]
priority this sprint is 'fiscal hosts can add pending contributions'

- [ ]

**Comment by kewitz** [comment]
- [X] Create project: https://github.com/opencollective/opencollective/projects/84
- [ ] Design to give a look at the existing wireframe
- [ ] Start working on CreatePendingContribution modal

**Comment by Memo-Es** [comment]
### Week 4

 - [ ] Review design together with @natehn @poohlaga and @iamronen

**Comment by iamronen** [comment]
Noting some comments on the [latest design](https://www.figma.com/file/KinuveYf4WUlRIXzMTnmIsmO/[OC-Design]-Lab-%F0%9F%A7%AA?node-id=18944%3A267097):

1. Dividing the form into clear sections: 
    - from (contributor information) 
    - to (collective, project/event, tier)
    - amount (amount, fees, purchase order, payment method, expected when)  + clarify here if fees are added on top of the amount or extracted from it. 
2. Payment method options: unknown, bank transfer, check.
3. Include a possibility to select an event or project to which the funds are to be directed within the collective.
5. Include a Purchase Order field (in the "amount" section).
6. Make it easy (instead of entering a date) to indicate when the funds are expected: within n months or n weeks.
7. Title ( @natehn what is your take on this? )     - to me this is an unclear field and creates a micro-tension: what should I put in here? 
    - I cannot think of where this description would be useful, where could it be surfaced in a way that could be helpful 
    - if we keep it, it should be optional 
    - it can be placed at the end and auto filled as the form is filled so that when you arrive at it there is a default title,eg: $50k + $5k host fees from contributor name to collective name - project/event name
    - the user can then edit or just do nothing and benefit from the default.
 7. Description field seems redundant (there are already notes and a perhaps redundant title).

**Comment by BenJam** [comment]
- [ ] review design changes (tomorrow)
- [ ] look at notifications that need to be generated
- [ ] look at create order page (for use later)

**Comment by kewitz** [comment]
- [ ] Submit for Review and Merge "Pending Contributions" page + CreatePendingContributionModal
- [ ] Submit Pending Contributions Email notifications for review
- [ ] Start Contribution page (single order information page)

**Comment by kewitz** [comment]
- [ ] Add Pending Contribution details to Pending Contributions list
- [ ] Finish and Submit Contribution/Order page for review
- [ ] List Pending Contributions on Collective transactions for Collective Admins
- [ ] Start Documentation

**Comment by jfdh** [comment]
Hi! I am jumping in here to propose a potential expansion of this feature as I've just used the _create +_ button.

I believe the pending contributions feature should help us reflect on how we are managing external invoicing for fiscal hosts. Especially in the EU where some of our biggest collectives are incorporated and thus able to invoice clients to get paid for a service by using OCE tax data. 



### User story

As a fiscal host administrator / operator, collectives often ask us if they can invoice their clients. Some collectives are selling services (i.e. an artistic performance, a speech at a conference). This is a sensitive topic fiscally wise and we feel like the host admins should be the only one to actually send orders to the outside world for the collectives. We do not want to allow collectives to send invoices to clients using OCE data without the fiscal host approval. As far as OCE is concerned, the pending contribution feature _should not_ be available to our collectives but only on the Host Dashboard.

Here is how we are sending invoices to clients for the collectives: 

We ask the collective to send us the data they wanna have in the invoice (Identification of the client, VAT rate, price - nature of the service,...). And then we do an invoice on Google Sheets and send it to the client of the collective. This is a very time consuming (and not automated) task that I feel is the last big piece missing to Open Collective to have a coherent system of in / out funds management. 

Obviously, no order is being automatically created when going through this process and we are adding the funds manually to the collectives.

For example: DemocracyNext collective organises workshop and are being paid for that. They would not be able to do so without OCE VAT number and us creating invoices for them so that they can send it to their clients (i.e. a museum or another cultural institutions).

Here is the transaction on their collective that shows a payment they receive because the client paid this invoice (that we manually created for them) on their collective.

### Best solution for this problem

A potential expansion of the pending contributions and the _create +_ button **for Fiscal Host Admins** (not for collective) could be:

1. A comprehensive identification of the client and of the object of the order: Enhance the data fields in the _create +_ button (of the host admin dashboard) to match the custom needs of each fiscal host (i.e. add the possibility to add VAT, or indicate EU tax reverse charge, the precise identification of the client, the item)
2. A notification to the client: when the host admin is confirming the order on the dashboard for the collective, the client should receive in his mailbox an invitation to pay the pending order including an attachment with the invoice.

I am happy to discuss this in a call with whom it might be best relevant. This is something that could really improve drastically the collective's user experience as well as help our operations a lot. 

Thanks for your consideration!

**Comment by iamronen** [comment]
Hey @jfdh thank you for elaborating on your need. 

I would like to try unpacking this into a few separate issues here:
1. An ability to account for VAT when receiving funds. Once we have a clear approach on how to systemically handle VAT we should be able to revisit the adding funds capability to accommodate for that.
2. Once we account for VAT the pending contributions feature should serve you well in conjunction with an external invoicing tool. You create the invoices on an invoicing tool of your choice and then you can create pending contributions to match them for tracking on the platform for incoming funds. 
3. It is clear that issuing invoices is a recurring need, and it is also clear that we do not yet know if/how we can/should get involved in this. 
4. Your request to be able to add line-items may seem like a simple ask, but I think it can easily lead to added complexity that we should only take on based on a clear choice to make this a priority. This seemingly simple request, is an additional reminder that we should be very careful about opening the potential Pandora box of invoices.
5. I am also wondering if "Invited Expenses" could play a role in at least some of the cases?
6. If invoices DO NOT become a priority to solve on the platform in the foreseeable future, it may be possible for you to create automation the other way around: to create an interface from your invoicing tool of choice to automatically create pending contributions for you on the platform (through the API) so that you can track and reconcile the funding when it arrives.

**Comment by BenJam** [comment]
Note that @iamronen and I have spoken with @mfix-stripe and Dawn who have been using this at Open Collective Foundation. I think. this project still needs:

- [ ] documentation
- [ ] release 
- [ ] dashboard

we should decide then whether we want/need to follow up, given the usage that we have seen has been more like a document store than a tool for expenses payments.
